test.sh

```diff
@@ -1,43 +1,24 @@
 #!/bin/bash -ex
 
-test_image=api-test
-docker build -t $test_image .
-
 function finish {
-	docker rm -f $pg_cid
-	docker rm -f $server_cid
+  docker-compose down
 }
 trap finish EXIT
 
-possum_tag=0.1.0-stable
-possum_image=registry.tld/possum:$possum_tag
-
-export POSSUM_DATA_KEY="$(docker run --rm $possum_image data-key generate)"
-
-pg_cid=$(docker run -d postgres:9.3)
-
-server_cid=$(docker run -d \
-	--link $pg_cid:pg \
-	-e DATABASE_URL=postgres://postgres@pg/postgres \
-	-e RAILS_ENV=test \
-	$possum_image server)
+# Generate reports folders locally
+mkdir -p spec/reports features/reports
 
-set -o pipefail
-admin_api_key=( $(cat ci/setup-account.sh | docker exec -i $server_cid /bin/bash | tail -1) )
+# Build test container & start the cluster
+docker-compose build
+docker-compose up -d
 
-mkdir -p spec/reports features/reports
+# Delay to allow time for Possum to come up
+# TODO: remove this once we have HEALTHCHECK in place
+sleep 20
 
-docker run \
-	-i \
-	--rm \
-	--link $pg_cid:pg \
-	--link $server_cid:possum \
-    -v $PWD/spec/reports:/src/conjur-api/spec/reports \
-    -v $PWD/features/reports:/src/conjur-api/features/reports \
-	-e DATABASE_URL=postgres://postgres@pg/postgres \
-	-e RAILS_ENV=test \
-	-e CONJUR_APPLIANCE_URL=http://possum \
-	-e CONJUR_ACCOUNT=cucumber \
-    -e CONJUR_AUTHN_API_KEY=${admin_api_key[3]} \
-    $test_image ci/test.sh "$@"
+api_key=$(docker-compose exec -T possum rails r "print Credentials['cucumber:user:admin'].api_key")
 
+# Execute tests
+docker-compose exec -T tests \
```

Comment:
I think we would be better off using a `run` here, since as we know `exec` is a box-full of knives.

```
docker-compose run --rm \
  -e CONJUR_AUTHN_API_KEY="$api_key" \
  tests bash -c 'ci/test.sh'
```

That way, you don't need to worry about cleanup either.

Jenkins is already garbage-collecting our old images so you don't have to worry about them as much as containers.